self-tests-staging.yml

```diff
@@ -22,7 +22,7 @@ buildvariants:
       goarch: amd64
       IS_DOCKER: true
       GOROOT: /usr/local/go
-      mongodb_url: https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-ubuntu1604-4.2.18.tgz
+      mongodb_url: https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-ubuntu1604-4.4.18.tgz
```

Comment:
We are currently running 5.0 everywhere else.
https://github.com/evergreen-ci/evergreen/blob/39e5eb530789706fb6147b0a8465928ad18fcdae/self-tests.yml#L622

Soon it will be 6.0 https://jira.mongodb.org/browse/EVG-18898

Reply:
Ok, changed the scope of this ticket to upgrade the whole container variant to use a Ubuntu 20.04 image, which supports Mongo v5, and changed this url to v5 as well. Confirmed tests are still passing.

Reply:
Thanks! LGTM